anvil/src/drawing.rs

```diff
@@ -178,6 +178,78 @@ where
     result
 }
 
+fn draw_popup<R, E, F, T>(
+    renderer: &mut R,
+    frame: &mut F,
+    egl_buffer_reader: Option<&EGLBufferReader>,
+    compositor_token: MyCompositorToken,
+    surface: &wl_surface::WlSurface,
+    log: &Logger,
+) -> Result<(), SwapBuffersError>
+where
```

Comment:
This function does not handle potential subsurfaces of popup surfaces. Is there a specific reason for not using `draw_surface_tree` for drawing the popups as well?

Reply:
Popup drawing has been removed from the PR as discussed on matrix. I will try to implement drawing in a separete PR.